### PR TITLE
Waypoint/Routes editable text color and font

### PR DESCRIPTION
**Pull request by domi4484**
- Editable label font and color for Waypoints and Routes
- Waypoint/Route KML export use name as placeholder name
- Waypoint KML export placeholder size, color, label color and label size

Peek 2024-11-15 11-36

### CODE REVIEW

File: kadas/gui/mapitemeditors/kadasgpxrouteeditor.cpp [hunk 1]
```diff
@@ -122,3 +173,21 @@ void KadasGpxRouteEditor::saveSize()
   QgsSettings().setValue( "/gpx/route_size", mUi.mSpinBoxSize->value() );
   emit styleChanged();
 }
+
+void KadasGpxRouteEditor::saveLabelFont()
+{
+  QgsSettings().setValue( "/gpx/route_label_font", currentFont().toString() );
```
Comment (3nids) [inline kadas/gui/mapitemeditors/kadasgpxrouteeditor.cpp:179]: can you use new settings API?

File: kadas/gui/mapitemeditors/kadasgpxwaypointeditor.cpp [hunk 2]
```diff
@@ -96,3 +144,21 @@ void KadasGpxWaypointEditor::saveSize()
   QgsSettings().setValue( "/gpx/waypoint_size", mUi.mSpinBoxSize->value() );
   emit styleChanged();
 }
+
+void KadasGpxWaypointEditor::saveLabelFont()
+{
+  QgsSettings().setValue( "/gpx/waypoint_label_font", currentFont().toString() );
```
Comment (3nids) [inline kadas/gui/mapitemeditors/kadasgpxwaypointeditor.cpp:150]: idem

File: kadas/gui/mapitems/kadasgpxwaypointitem.cpp [hunk 3]
```diff
@@ -69,15 +92,86 @@ void KadasGpxWaypointItem::render( QgsRenderContext &context ) const
...
+    pos += QPointF( 0.5 * mIconSize, -0.5 * mIconSize );
+    QColor bufferColor = ( 0.2126 * mIconBrush.color().red() + 0.7152 * mIconBrush.color().green() + 0.0722 * mIconBrush.color().blue() ) > 128 ? Qt::black : Qt::white;
+
+    QFont font = mLabelFont;
+    font.setPointSizeF( font.pointSizeF() * outputDpiScale( context ) );
+    QFontMetrics metrics( font );
+    pos += QPointF( metrics.width( mName ) / 2.0, 0.0 );
+
+    // no idea why this works, otherwise text scales up when edited
+    // the rendex context is coming from KadasMapItem when edited while it comes from the QgsMapLayerRenderer otherwise
```
Comment (3nids) [inline kadas/gui/mapitems/kadasgpxwaypointitem.cpp:105]: looks familiar ;) 

shall we move this to some utils class to avoid code duplication?

Comment (domi4484) [reply kadas/gui/mapitems/kadasgpxwaypointitem.cpp]: Yes I think you spared me some hours of struggling here!

I put that in the parent class

File: kadas/gui/mapitemeditors/kadasgpxrouteeditor.cpp (hunk 1 above)
Comment (domi4484) [reply kadas/gui/mapitemeditors/kadasgpxrouteeditor.cpp]: Done, but old value for colors is not taken over correctly. Do you use some tricks or migration code for those?

**Pull request merged**